skimage/segmentation/_chan_vese.py

```diff
@@ -0,0 +1,289 @@
+    (c1, c2) = _cv_calculate_averages(img, Hphi)
+    Hinv = 1.-Hphi
+    return lambda_pos*(img-c1)**2*Hphi+lambda_neg*(img-c2)**2*Hinv
+
+
+def _cv_edge_length_term(phi, mu):
+    """
+    Returns the 'energy' contribution due to the length of the
+    edge between regions at each point, multiplied by a factor 'mu'.
+    """
+    toret = _cv_curvature(phi)
+    return mu*toret
+
+
+def _cv_energy(img, phi, mu, lambda1, lambda2, heavyside=_cv_heavyside):
+    """
+    Returns the total 'energy' of the current level set function
+    """
+    H = heavyside(phi)
+    return np.sum(_cv_difference_from_average_term(img, H, lambda1, lambda2) +
+                  _cv_edge_length_term(phi, mu))
+
+
+def _cv_reset_level_set(phi):
+    """
+    This is a placeholder function as resetting the level set is not
+    strictly necessary, and has not been done for this implementation.
+    """
+    return phi
+
+
+def _cv_checkerboard(image_size, square_size):
+    """
+    Generates a checkerboard level set function. According to
+    Pascal Getreuer, such a level set function has fast convergence.
+    """
+    yv = np.arange(image_size[0]).reshape(image_size[0], 1)
+    xv = np.arange(image_size[1])
+    return (np.sin(np.pi/square_size*yv)*(np.sin(np.pi/square_size*xv)))
+
+
+def _cv_large_disk(image_size):
+    """
+    Generates a disk level set function. The disk covers the whole
+    image along its smallest dimension.
+    """
+    res = np.ones(image_size)
+    centerY = int((image_size[0]-1)/2)
+    centerX = int((image_size[1]-1)/2)
+    res[centerY, centerX] = 0.
+    radius = float(min(centerX, centerY))
+    return (radius-distance(res))/radius
+
+
+def _cv_small_disk(image_size):
+    """
+    Generates a disk level set function. The disk covers half of
+    the image along its smallest dimension.
+    """
+    res = np.ones(image_size)
+    centerY = int((image_size[0]-1)/2)
+    centerX = int((image_size[1]-1)/2)
+    res[centerY, centerX] = 0.
+    radius = float(min(centerX, centerY))
+    return (radius/2-distance(res))/(radius*1.5)
+
+
+def _cv_initial_shape(starting_level_set, img):
+    """
+    Generates a checkerboard level set function. According to
+    Pascal Getreuer, such a level set function has fast convergence.
+    """
+    res = []
+    if type(starting_level_set) == str:
+        if starting_level_set == 'checkerboard':
+            res = _cv_checkerboard(img.shape, 5)
+        elif starting_level_set == 'disk':
+            res = _cv_large_disk(img.shape)
+        elif starting_level_set == 'small disk':
+            res = _cv_small_disk(img.shape)
+        else:
+            raise ValueError("Incorrect name for starting level set preset.")
+    else:
+        res = starting_level_set
+    return res
+
+
+def chan_vese(img, mu, lambda1=1.0, lambda2=1.0, tol=1.0, maxiter=100, dt=1.0,
+              starting_level_set='checkerboard', extended_output=False):
+    """Returns a bool array representing two separate regions
```

Comment:
I'd say let the `returns` section document what is returned, and make the top level comment similar to: https://github.com/blink1073/scikit-image/blob/master/skimage/segmentation/active_contour_model.py#L13.